### PR TITLE
Add StratisCertify tests for snapshot revert

### PR DESCRIPTION
**Pull request by bgurney-rh**
Related: #280

### CODE REVIEW

**Comment by bgurney-rh** [comment]
The cancel-revert test seems like the only test that I can add right now, as the others would have remnants that clean_up() can't handle right now:
```
Traceback (most recent call last):
- File "testlib/infra.py", line 126, in clean_up
    f'remnant pools: {", ".join(name for name in remnant_pools)}'
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: sequence item 0: expected str instance, tuple found
```

...and I had to manually run "stratis fs cancel-revert" for the remnant filesystem snapshot.

**Comment by mulkieran** [comment]
We'll need to augment the cleanup method with a step to unset all merge fields before we can proceed with the stratis-cli tests.

**Comment by bgurney-rh** [comment]
I found a way to get the clean_up() method to run, but it took some rework.  Re-ordered commits to have this rework done first, before adding the first test.

**Comment by bgurney-rh** [comment]
Not quite working yet; for some reason, the `clean_up()` method is not setting "merge" to "false" on the test filesystem created as a result of the "stratis fs snapshot" command.

**Comment by bgurney-rh** [comment]
Rebasing, putting all of the testlib additions/fixes in the first commit.

File: testlib/infra.py [hunk 1]
```diff
@@ -89,8 +89,22 @@ def check_result(result, format_str, format_str_args):
                     f"{os.path.join(VAR_TMP, mountpoint_dir, name)}: {err}"
                 )
 
+    # Unset MergeScheduled
+    for (fs_path, name, origin), pool_name in StratisDbus.fs_list().items():
```
Comment (mulkieran) [inline testlib/infra.py:93]: Here, I think you must as well destructure more, like:

```
for (fs_path, name, (origin_set, _)), pool_name in StratisDbus.fs_list().items():
  if origin_set:
    ...
```

**Review by mulkieran — CHANGES_REQUESTED** [review]
I have just one requested change. You might as well go on and add a stratis_cert.py test as well and then this should be done.

File: testlib/infra.py (hunk 1 above)
Comment (bgurney-rh) [reply testlib/infra.py]: OK, should be resolved.

**Comment by mulkieran** [comment]
Looks good

**Comment by bgurney-rh** [comment]
Rebased, to squash  `clean_up(): destructure MergeScheduled unset check` into the main commit for `clean_up()` changes.

**Comment by bgurney-rh** [comment]
Rebased.

**Review by mulkieran — APPROVED** [review]

**Pull request merged**